draft-ietf-wish-whep.md

```diff
@@ -75,27 +75,33 @@ The following diagram illustrates the core operation of the WHEP protocol for in
 
 ~~~~~
                                                                                
- +-------------+    +---------------+ +--------------+ +---------------+
- | WHEP player |    | WHEP endpoint | | Media Server | | WHEP session |
- +--+----------+    +---------+-----+ +------+-------+ +--------|------+
-    |                         |              |                  |       
-    |                         |              |                  |       
-    |HTTP POST (SDP Offer)    |              |                  |       
-    +------------------------>+              |                  |       
-    |201 Created (SDP answer) |              |                  |       
-    +<------------------------+              |                  |       
-    |          ICE REQUEST                   |                  |       
-    +--------------------------------------->+                  |       
-    |          ICE RESPONSE                  |                  |       
-    |<---------------------------------------+                  |       
-    |          DTLS SETUP                    |                  |       
-    |<======================================>|                  |       
-    |          RTP/RTCP FLOW                 |                  |       
-    +<-------------------------------------->+                  |       
-    | HTTP DELETE                                               |       
-    +---------------------------------------------------------->+       
-    | 200 OK                                                    |       
-    <-----------------------------------------------------------x       
+ +-------------+            +---------------+ +--------------+ +---------------+
+ | WHEP player |            | WHEP endpoint | | Media Server | | WHEP session |
+ +--+----------+            +---------+-----+ +------+-------+ +--------|------+
+    |                                 |              |                  |       
+    |                                 |              |                  |       
+    |HTTP POST (Empty body)           |              |                  |       
```

Comment:
Tim P mentioned he didn't like the whole "empty post" thing... 

So lets talk about that in this draft too.

A HTTP POST should create something, which this does.
A HTTP GET is meant to get an already existing resource, in this case the resource doesn't exist yet so you can't go get it...
PATCH, HEAD, OPTIONS, DELETE don't make sense here.

And so I think HTTP POST with an empty body works...

Reply:
Empty POST works perfectly fine in theory and practice, at least in the cases I've run through. Since a POST is expected in either case / mode, its easy to follow from a processing stand-point.

Reply:
Well, yes, it works, but the fact that we now require an offer-answer exchange that has 4 messages and 2 of them are empty tells you everything you need to know about the elegance of this method. (I admit the original SDP design is really to blame).